src/Repository/RepositoryMapUpdater.php

```diff
@@ -0,0 +1,150 @@
+<?php declare(strict_types=1);
+
+namespace OAT\DependencyResolver\Repository;
+
+use OAT\DependencyResolver\Repository\Entity\Repository;
+use OAT\DependencyResolver\Repository\Interfaces\RepositoryReaderInterface;
+use Packagist\Api\Client as PackagistReader;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+
+class RepositoryMapUpdater implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    /** @var RepositoryMapAccessor */
+    private $repositoryMapAccessor;
+
+    /** @var RepositoryReaderInterface */
+    private $repositoryReader;
+
+    /** @var PackagistReader */
+    private $packagistReader;
+
+    /**
+     * RepositoryMapUpdater constructor.
+     *
+     * @param RepositoryMapAccessor     $repositoryMapAccessor
+     * @param RepositoryReaderInterface $repositoryReader
+     * @param PackagistReader           $packagistReader
+     */
+    public function __construct(
+        RepositoryMapAccessor $repositoryMapAccessor,
+        RepositoryReaderInterface $repositoryReader,
+        PackagistReader $packagistReader
+    ) {
+        $this->repositoryMapAccessor = $repositoryMapAccessor;
+        $this->repositoryReader = $repositoryReader;
+        $this->packagistReader = $packagistReader;
+    }
+
+    /**
+     * Reads the repositories name of each repository in the list
+     *
+     * @param string $userName   Owner of the repositories
+     * @param bool   $reloadList Do we need to reload list of repositories?
+     * @param int    $limit      Number of repositories names to read
+     */
+    public function update(string $userName, bool $reloadList, int $limit)
+    {
+        // Number of updated repositories.
+        $updated = 0;
+        // Number of skipped repositories.
+        $skipped = 0;
+
+        // Reload repositoryList to get repositories not mapped yet.
+        if ($reloadList) {
+            $this->logger->info($this->reloadList($userName) . ' repositories added.');
+        }
+
+        // Reads either local or distant repository list.
+        $repositoryList = $this->repositoryMapAccessor->read();
+
+        // Displays the number of repositories to update.
+        $toUpdate = 0;
+        foreach ($repositoryList as $repository) {
+            /** @var Repository $repository */
+            if ($repository->getExtensionName() === '') {
+                $toUpdate++;
+            }
+        }
+        $this->logger->info(
+            $toUpdate . ' repositories(s) to analyze' . ($limit ? ' (limited to ' . $limit . ')' : '') . '.'
+        );
+
+        // Finds all repository that have an empty extension name.
+        foreach ($repositoryList as $repositoryName => &$repository) {
```

Comment:
Do not pass parameter by reference, it's not clean. Also you are iterating through the repositories twice, please refactor

Reply:
Refactored